ethdb/privateapi/ethbackend.go

```diff
@@ -270,6 +285,56 @@ func (s *EthBackendServer) EngineGetPayloadV1(ctx context.Context, req *remote.E
 	return nil, fmt.Errorf("unknown payload")
 }
 
+// EngineForkChoiceUpdatedV1, either states new block head or request the assembling of a new bloc
+func (s *EthBackendServer) EngineForkChoiceUpdatedV1(ctx context.Context, req *remote.EngineForkChoiceUpdatedRequest) (*remote.EngineForkChoiceUpdatedReply, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.config.TerminalTotalDifficulty == nil {
+		return nil, fmt.Errorf("not a proof-of-stake chain")
+	}
+	// Check if parent equate to the head
+	parent := gointerfaces.ConvertH256ToHash(req.Forkchoice.HeadBlockHash)
+	tx, err := s.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	headHeader, err := rawdb.ReadHeaderByHash(tx, parent)
+	if err != nil {
+		return nil, err
+	}
+
+	if atomic.LoadUint32(s.waitingForBeaconChain) == 0 || headHeader == nil {
+		return &remote.EngineForkChoiceUpdatedReply{
+			Status: string(Syncing),
+		}, nil
+	}
+
+	// Hash is incorrect because mining archittecture has yet to be implemented
```

Comment:
Hash can be calculated from the header. Theoretically we have enough data to construct a valid empty block. Since there are no transactions and no block rewards (after the Merge), state root doesn't change. We also would need to calculate the base fee per EIP-1559 rules.
On the other hand, perhaps it's not worth it to bother since we're going to add proper PoS mining with transactions in a subsequent PR.

Reply:
yup, that was my point